lib/library.rb

```diff
@@ -0,0 +1,45 @@
+class Library
+  attr_reader :authors, :books, :name, :checked_out_books
+
+  def initialize(name)
+    @name = name
+    @authors = []
+    @books = []
+    @checked_out_books = []
+    @books_by_popularity = Hash.new(0)
+  end
+
+  def add_author(author)
+    @authors << author
+    author.books.map {|book| @books << book}
```

Comment:
What if the author writes another book after the author has been added to the library? Then how would you get that book in?? :)